data/lib/src/datasource/authentication_datasource.dart

```diff
@@ -53,30 +54,30 @@ class AuthenticationDataSource {
           PermanentTokenBodyRequest('LinShare-${deviceManager.getPlatformString()}-$deviceUUID'));
       return permanentToken.toToken();
     }).catchError((error) {
-      _throwAuthenticateException(error);
+        _remoteExceptionThrower.throwRemoteException(error, handler: (DioError error) {
+          if (error.response.statusCode == 401) {
+            throw BadCredentials();
+          } else if (error.response.statusCode == 500) {
+            throw MissingRequiredFields();
+          } else {
+            throw UnknownError(error.response.statusMessage);
+          }
+        });
     });
   }
 
-  void _throwAuthenticateException(dynamic exception) {
-    if (exception is DioError) {
-      switch (exception.type) {
-        case DioErrorType.DEFAULT:
-          throw ServerNotFound();
-          break;
-        case DioErrorType.CONNECT_TIMEOUT:
-          throw ConnectError();
-          break;
-        case DioErrorType.RESPONSE:
-          if (exception.response.statusCode == 401) {
-            throw BadCredentials();
+  Future<bool> deletePermanentToken(Token token) async {
+    return await linShareHttpClient.deletePermanentToken(token.toPermanentToken())
+      .catchError((error) =>
+        _remoteExceptionThrower.throwRemoteException(error, handler: (DioError error) {
```

Comment:
also print this error to get the detail problem here

Reply:
it is cool if you have the response body in error

Reply:
`I/flutter ( 6701): datasourceDioError [DioErrorType.RESPONSE]: Http status error [415]`

response body is blank on my side

Reply:
have tested to add application/json to headers but it's not working